bundles/org.eclipse.jface.text/src/org/eclipse/jface/internal/text/html/HTMLPrinter.java

```diff
@@ -63,6 +63,11 @@ private static void installColorUpdater(final Display display) {
 	}
 
 	private static org.eclipse.text.html.RGB fromRGB(RGB val) {
+		// Preserve	null RGB as HTMLBuilder contains the default colors and sets them accordingly
```

Comment:
The problem here is the second public constructor that initializes everything **without** null check. Means, this would not work for client code that creates `HTMLBuilder(RGB bg, RGB fg, RGB link, RGB alink)` with some "null" color.
I haven't reviewed original PR that introduced this constructor, and it should not be used yet widely. So I think this PR should add some assert in that constuctor.

Reply:
The NPE seen happens before even getting into HTMLBuilder but I'll add guards for that too.

Reply:
No null colors now with latest revision.